core/src/main/java/jenkins/I18n.java

```diff
@@ -93,7 +93,16 @@ public HttpResponse doResourceBundle(StaplerRequest request) {
         String language = request.getParameter("language");
         String country = request.getParameter("country");
         String variant = request.getParameter("variant");
-
+        // https://www.w3.org/International/questions/qa-lang-priorities
+        // in case we have regions/countries in the language query parameter
+        if (country == null && language.length()>2) {
+            country = language.substring(3,5);
+            // not mentioned in above url but we assume variants as en-US-Cloudbees
+            if (variant == null && language.length()>5) {
+               variant = language.substring(6);
```

Comment:
What is at and after index 6? Maybe it would be good to provide some strategic examples of the strings you are parsing i.e. at the appropriate place state an example of the type of language string it is expecting to parse.